_posts/05-05-01-PHP-and-UTF8.md

```diff
@@ -99,20 +105,26 @@ $link = new PDO(
  
 // Store our transformed string as UTF-8 in our database
 // Your DB and tables are in the utf8mb4 character set and collation, right?
-$handle = $link->prepare('insert into ElvishSentences (Id, Body) values (?, ?)');
-$handle->bindValue(1, 1, PDO::PARAM_INT);
-$handle->bindValue(2, $string);
+$handle = $link->prepare('insert into ElvishSentences (Id, Body, Priority) values (default, ?, ?)');
+$handle->bindValue(1, $string); // Insert the first parameter
+$handle->bindValue(2, $priority=45, PDO::PARAM_INT); // For second parameter, explicitly tell pdo to expect an int
 $handle->execute();
  
 // Retrieve the string we just stored to prove it was stored correctly
 $handle = $link->prepare('select * from ElvishSentences where Id = ?');
-$handle->bindValue(1, 1, PDO::PARAM_INT);
+$handle->bindValue(1, $id=1, PDO::PARAM_INT);
```

Comment:
Sure, updated to that approach.

On Wed, Jan 7, 2015 at 3:49 PM, Phil Sturgeon <notifications@github.com>
wrote:

> In _posts/05-05-01-PHP-and-UTF8.md
> <https://github.com/codeguy/php-the-right-way/pull/529#discussion-diff-22614667>
> :
>
> >  $handle->execute();
> >
> >  // Retrieve the string we just stored to prove it was stored correctly
> >  $handle = $link->prepare('select * from ElvishSentences where Id = ?');
> > -$handle->bindValue(1, 1, PDO::PARAM_INT);
> > +$handle->bindValue(1, $id=1, PDO::PARAM_INT);
>
> An you just shove the IDs somewhere else, then reference them in the PDO
> code?
> --  Phil From: Roy R. <notifications@github.com>
> Reply: codeguy/php-the-right-way <reply@reply.github.com>> Date: January
> 7, 2015 at 3:47:13 PM To: codeguy/php-the-right-way <
> php-the-right-way@noreply.github.com>> Cc: Phil Sturgeon <
> me@philsturgeon.uk>> Subject:  Re: [php-the-right-way] PDO binding with
> utf-8: Clarified some of the code, and use escape function. (#529
> <https://github.com/codeguy/php-the-right-way/pull/529>) In
> _posts/05-05-01-PHP-and-UTF8.md:
>  $handle->execute(); // Retrieve the string we just stored to prove it was
> stored correctly $handle = $link->prepare('select * from ElvishSentences
> where Id = ?'); -$handle->bindValue(1, 1, PDO::PARAM_INT);
> +$handle->bindValue(1, $id=1, PDO::PARAM_INT);
> Yes indeed, sir, I added those temporary variables to clarify what was
> being done with the PDO bindings, since $handle->bindValue(1, 1,
> PDO::PARAM_INT); is almost pure jargon with magic numbers, and I don't
> think it's very clear to those unused to using numbered parameter bindings
> in pdo. I suppose one way to clarify that example would be to move to named
> parameters instead of numeric parameters (which is what I tend to use
> everywhere for the clarity it brings). Just let me know if you'd prefer
> that. — Reply to this email directly or view it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codeguy/php-the-right-way/pull/529/files#r22614667>.
>



-- 
+++++++++++++++++++++++++++
BitLucid.com <http://bitlucid.com/> - BitLucid, Inc. - Web & Programming
Consultants
ninjawars.net - My webgame project
github.com/tchalvak/ninjawars - My open source projects
dnaexmosn.deviantart.com - Art Gallery
585-519-7658  - Cell
"Never compare your inside with somebody's outside."